tests/pathfinding/test_service.py

```diff
@@ -106,17 +107,137 @@ def new_service(filename):
+        token_network_address=token_network_model.address,
+        channel_identifier=ChannelID(1),
+        participant1=participant1,
+        participant2=participant2,
+        settle_timeout=20,
+        block_number=BlockNumber(1),
+    )
+
+    assert len(pathfinding_service_mock.token_networks) == 1
+    assert len(token_network_model.channel_id_to_addresses) == 0
+
+    pathfinding_service_mock.handle_event(channel_event)
+    assert len(pathfinding_service_mock.token_networks) == 1
+    assert len(token_network_model.channel_id_to_addresses) == 1
+
+    # Test invalid token network address
+    channel_event.token_network_address = TokenNetworkAddress("0x" + "0" * 40)
+
+    pathfinding_service_mock.handle_event(channel_event)
+    assert len(pathfinding_service_mock.token_networks) == 1
+    assert len(token_network_model.channel_id_to_addresses) == 1
+
+
+def test_token_channel_new_deposit(pathfinding_service_mock, token_network_model):
+    participant1 = Address("0x" + "1" * 40)
+    participant2 = Address("0x" + "2" * 40)
+    channel_event = ReceiveChannelOpenedEvent(
+        token_network_address=token_network_model.address,
+        channel_identifier=ChannelID(1),
+        participant1=participant1,
+        participant2=participant2,
+        settle_timeout=20,
+        block_number=BlockNumber(1),
+    )
+
+    pathfinding_service_mock.handle_event(channel_event)
+    assert len(pathfinding_service_mock.token_networks) == 1
+    assert len(token_network_model.channel_id_to_addresses) == 1
+
+    deposit_event = ReceiveChannelNewDepositEvent(
+        token_network_address=token_network_model.address,
+        channel_identifier=ChannelID(1),
+        participant_address=participant1,
+        total_deposit=TokenAmount(123),
+        block_number=BlockNumber(2),
+    )
+
+    pathfinding_service_mock.handle_event(deposit_event)
+    assert len(pathfinding_service_mock.token_networks) == 1
+    assert len(token_network_model.channel_id_to_addresses) == 1
+
+    # Test invalid token network address
+    deposit_event.token_network_address = TokenNetworkAddress("0x" + "0" * 40)
+
+    pathfinding_service_mock.handle_event(deposit_event)
+    assert len(pathfinding_service_mock.token_networks) == 1
+    assert len(token_network_model.channel_id_to_addresses) == 1
+
+
+def test_token_channel_closed(pathfinding_service_mock, token_network_model):
+    participant1 = Address("0x" + "1" * 40)
+    participant2 = Address("0x" + "2" * 40)
+    channel_event = ReceiveChannelOpenedEvent(
+        token_network_address=token_network_model.address,
+        channel_identifier=ChannelID(1),
+        participant1=participant1,
+        participant2=participant2,
+        settle_timeout=20,
+        block_number=BlockNumber(1),
+    )
+
+    pathfinding_service_mock.handle_event(channel_event)
+    assert len(pathfinding_service_mock.token_networks) == 1
+    assert len(token_network_model.channel_id_to_addresses) == 1
```

Comment:
This part is repeated three times. Reusing it would make the tests a bit shorter.